src/sqlparser/src/parser.rs

```diff
@@ -1943,6 +1969,41 @@ impl Parser {
         Ok(data_type)
     }
 
+    /// Parse struct columns e.g.`<v1 int, v2 int, v3 struct<...>>`.
+    pub fn parse_struct_columns(&mut self) -> Result<Vec<ColumnDef>, ParserError> {
```

Comment:
Can we just set all these struct fields in DataType? Because the definition of a struct field is far different from a column. A column can be optionally a PRIMARY KEY, or UNIQUE KEY, but apparently a struct is not supposed to have these properties.